specs/GetFavicon.md

```diff
@@ -0,0 +1,153 @@
+is fired when the page is navigated to. The Favicon would be an
+empty image and empty Uri. The user is expected to handle this scenario.
+Otherwise we fire the FaviconChanged with an observed change to the
+Favicon. In that scenario, we would provide the Uri of the Favicon we
+observed and a copy of the image.
+See [API Details](#api-details) Section below for API reference
+# API Details
+## Win32 C++
+```cpp
+/// This interface is a handler for when the `Favicon` is changed.
+/// The sender is the embedded_browser which handle the change and the
+/// second argument is null.
+/// For more information see `add_FaviconChanged`.
+[uuid(2913DA94-833D-4DE0-8DCA-900FC524A1A4), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalFaviconChangedEventHandler : IUnknown {
+  /// Called to notify the favicon changed.
+  HRESULT Invoke(
+      [in] ICoreWebView2* sender,
+      [in] IUnknown* args);
+}
+
+/// This interface is a handler for the completion of the copying for the`imageStream`.
+/// The 'error_code` is E_NOT_SET if the there is no image. Otherwise error_code 
+/// is the result from the image write operation.
+/// For more details, see the `GetFavicon` API.
+[uuid(A2508329-7DA8-49D7-8C05-FA125E4AEE8D), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalGetFaviconCompletedHandler : IUnknown {
+  /// Called to notify the favicon has been retrieved.
+  HRESULT Invoke([in] HRESULT error_code);
+}
+
+/// This is the ICoreWebView2 Experimental Favicon interface.
+/// It has the capability to fire an event for the changing of the
+/// Favicon. When a new page loads event would fire due to Favicon not being set.
+/// The event would also set the Favicon when the set by DOM or Javascript.
+[uuid(DC838C64-F64B-4DC7-98EC-0992108E2157), object, pointer_default(unique)]
+interface ICoreWebView2_10 : ICoreWebView2_9 {
+    /// Add an event handler for the `FaviconChanged` event.
+    HRESULT add_FaviconChanged(
+        [in] ICoreWebView2ExperimentalFaviconChangedEventHandler* eventHandler,
+        [out] EventRegistrationToken* token);
+
+    /// Remove the event handler for `FaviconChanged` event.
+    HRESULT remove_FaviconChanged(
+        [in] EventRegistrationToken token);
+
+    /// Get the current Uri of the favicon as a string.
+    /// If value is null, the `HRESULT` `E_POINTER`, otherwise it is `S_OK`.
+    /// If a page has no favicon then value is an empty string.
+    [propget] HRESULT FaviconUri([out, retval] LPWSTR* value);
+
+    /// Async function for getting the actual image data of the favicon.
+    /// If the `imageStream` is null, the `HRESULT` will be `E_POINTER`, otherwise
+    /// it is `S_OK`.
+    /// The image is copied to the `imageStream` object and when complete. If
+    /// there is no image then no data would be copied into the imageStream.
+    /// In either scenario the `eventHandler` is executed at the end of the operation.
```

Comment:
eventHandler should be named completedHandler or something like that. Not an event.